src/components/SearchForm/SearchForm.js

```diff
@@ -1,32 +1,44 @@
-import { useContext, useState } from "react";
+import { useContext, useEffect } from "react";
 import SearchNewsContext from "../../contexts/SearchNewsContext";
-
 import SavedNewsKeywordContext from "../../contexts/SavedNewsKeyword";
+import "./SearchForm.css";
 
 const SearchForm = ({ handleSearchNews }) => {
-  // const [searchInput, setSearchInput] = useState("");
   const { searchInput, setSearchInput } = useContext(SearchNewsContext);
-  const { savedKeyword } = useContext(SavedNewsKeywordContext);
-  console.log("savedKeyword", savedKeyword);
+  const { setSavedKeyword } = useContext(SavedNewsKeywordContext);
+
   const handleSearchInput = (e) => {
     setSearchInput(e.target.value);
   };
   const handleSearchSubmit = (e) => {
     e.preventDefault();
     handleSearchNews(searchInput);
   };
-  // console.log(searchInput);
+
+  useEffect(() => {
+    if (searchInput) {
+      setSavedKeyword(searchInput);
+    } else {
+      setSavedKeyword("");
+    }
+  }, [searchInput, setSavedKeyword]);
 
   return (
     <>
-      <form onSubmit={handleSearchSubmit} className="main__search-form">
+      <form onSubmit={handleSearchSubmit} className="searchform">
         <input
           type="text"
-          className="main__search-input"
+          className="searchform__input"
           placeholder="Nature"
           onChange={handleSearchInput}
```

Comment:
If you have `onChange` or ` onInput` in an input then you need to add attribute `value` to this input.
This is how React controls inputs.

To read more about **controlled components**: https://reactjs.org/docs/forms.html#controlled-components